cmake/toolchain_check.cmake

```diff
@@ -12,9 +12,10 @@ if(TARGET_CPU STREQUAL "x86_64")
     set(crt_lib "--disable-lib32 --enable-lib64")
     set(LIBOMP_ASMFLAGS_M64 "-DLIBOMP_ASMFLAGS=-m64")
     set(M_TUNE "generic")
-    set(cfi "-mguard=cf")
+    set(cfi "-mguard=cf -fcf-protection=full")
```

Comment:
I disagree adding more hardening other than mguard for general distribution. Plus there might be performance penalty for some cpu

Reply:
For your own cpu, its okay but for public release nope